apps/emqx_auth_jwt/src/emqx_auth_jwt.erl

```diff
@@ -75,14 +73,40 @@ check_acl(ClientInfo = #{jwt_claims := Claims},
     end.
 
 is_expired(Exp) when is_binary(Exp)  ->
-    ExpInt  = binary_to_integer(Exp),
-    is_expired(ExpInt);
-is_expired(Exp) ->
+    case binary_to_number(Exp) of
+        {ok, Val} ->
+            is_expired(Val);
+        _ ->
+            ?DEBUG("acl_deny_due_to_invalid_jwt_exp:~p", [Exp]),
+            true
+    end;
+is_expired(Exp) when is_integer(Exp) ->
     Now = erlang:system_time(second),
-    Now > Exp.
+    Now > Exp;
+is_expired(Exp) ->
+    ?DEBUG("acl_deny_due_to_invalid_jwt_exp:~p", [Exp]),
+    true.
 
 description() -> "Authentication with JWT".
 
+binary_to_number(Bin) ->
+    Checker = fun([], _) ->
```

Comment:
Nice use of fixed point combinator, but Erlang has named funs: https://www.erlang.org/eeps/eep-0037.html . Better yet, it can be moved to an internal top-level function.

Reply:
I don't know much about this feature, thank you very much